examples/app-with-backend/src/components/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState, ChangeEvent } from 'react';
+import { Button, Field, Input, useStyles2, FieldSet } from '@grafana/ui';
+import { PluginConfigPageProps, AppPluginMeta, PluginMeta, GrafanaTheme2 } from '@grafana/data';
+import { getBackendSrv, locationService } from '@grafana/runtime';
+import { css } from '@emotion/css';
+import { SecretInput } from '../SecretInput';
+import { testIds } from '../testIds';
+
+export type JsonData = {
+  apiUrl?: string;
+  isApiKeySet?: boolean;
+};
+
+type State = {
+  // The URL to reach our custom API.
+  apiUrl: string;
+  // Tells us if the API key secret is set.
+  // Set to `true` ONLY if it has already been set and haven't been changed.
+  // (We unfortunately need an auxiliray variable for this, as `secureJsonData` is never exposed to the browser after it is set)
+  isApiKeySet: boolean;
```

Comment:
Here we can make use of `secureJsonFields KeyValue<boolean>` to achieve this behaviour without the added field. This should exist on the PluginMeta. 

For reference the PR that added support: https://github.com/grafana/grafana/pull/55313